### PR TITLE
Whois bug fix

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/23331 
fixes https://github.com/demisto/etc/issues/22873

## Description
 - Removed duplicate fields from ***domain*** and ***whois*** commands output.
 - Added the *Domain.Whois.QueryValue* context output to ***whois*** command.
 - Added ability to query Whois with subdomains.  
 - Fixed an issue where Whois fails for too many requests.
 
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/Whois/Integrations/Whois/Whois.py [hunk 1]
```diff
@@ -7163,7 +7166,15 @@ def get_whois_raw(domain, server="", previous=None, rfc3490=True, never_cut=Fals
...
         request_domain = domain
-    response = whois_request(request_domain, target_server)
+    response = ""
+    for i in range(0, 3):
+        try:
+            response = whois_request(request_domain, target_server)
+        except socket.error as err:
+            if err.errno == errno.ECONNRESET:
+                continue
+            break
```
Comment (yaakovi) [inline Packs/Whois/Integrations/Whois/Whois.py:7176]: redundant

File: Packs/Whois/Integrations/Whois/Whois.py [hunk 2]
```diff
@@ -8238,10 +8251,23 @@ def get_whois(domain, normalized=None):
...
                            handle_server=server_list[-1])
 
+
 # Drops the mic disable-secrets-detection-end
 
+def get_domain_from_query(query):
+    # checks for largest matching suffix inside tlds dictionary
+    suffix_len = max([len(suffix) for suffix in tlds if query.endswith('.{}'.format(suffix))] or [0])
+    if suffix_len != 0:
+        suffix_len += 1
```
Comment (ronykoz) [inline Packs/Whois/Integrations/Whois/Whois.py:8261]: why?

File: Packs/Whois/Integrations/Whois/Whois.py [hunk 3]
```diff
@@ -8238,10 +8251,23 @@ def get_whois(domain, normalized=None):
     return parse_raw_whois(raw_data, normalized=normalized, never_query_handles=False,
                            handle_server=server_list[-1])
 
+
 # Drops the mic disable-secrets-detection-end
 
+def get_domain_from_query(query):
+    # checks for largest matching suffix inside tlds dictionary
+    suffix_len = max([len(suffix) for suffix in tlds if query.endswith('.{}'.format(suffix))] or [0])
```
Comment (ronykoz) [inline Packs/Whois/Integrations/Whois/Whois.py:8259]: Shouldn't this be:
```suggestion
    suffix_len = max([len(suffix) for suffix in tlds if query.endswith('.{}'.format(suffix))] or [])
```

File: Packs/Whois/Integrations/Whois/Whois.py [hunk 4]
```diff
@@ -8238,10 +8251,23 @@ def get_whois(domain, normalized=None):
     return parse_raw_whois(raw_data, normalized=normalized, never_query_handles=False,
                            handle_server=server_list[-1])
 
+
 # Drops the mic disable-secrets-detection-end
 
+def get_domain_from_query(query):
```
Comment (ronykoz) [inline Packs/Whois/Integrations/Whois/Whois.py:8257]: why not use libraries that to this for you?

File: Packs/Whois/Integrations/Whois/README.md [hunk 5]
```diff
@@ -0,0 +1,415 @@
+Provides data enrichment for domains.
+This integration was integrated and tested with version xx of Whois
```
Comment (ronykoz) [inline Packs/Whois/Integrations/Whois/README.md:2]: fill in the version

File: Packs/Whois/Integrations/Whois/Whois.py (hunk 4 above)
Comment (Bargenish) [reply Packs/Whois/Integrations/Whois/Whois.py:8257]: I've checked 2 libraries (one of them was `tldextract`), they weren't as accurate as expected so I did it manually.

File: Packs/Whois/Integrations/Whois/Whois.py (hunk 3 above)
Comment (Bargenish) [reply Packs/Whois/Integrations/Whois/Whois.py:8259]: No, tried it. max function in python 2.7 doesn't support an empty list. and doesn’t have a 'default' keyword parameter.

File: Packs/Whois/Integrations/Whois/Whois.yml [hunk 6]
```diff
@@ -232,48 +193,6 @@ script:
     - contextPath: Domain.Name
```
[the remark this answers is not in the document]
Comment (Bargenish) [reply Packs/Whois/Integrations/Whois/Whois.yml:193]: The third issue in https://github.com/demisto/etc/issues/22873

File: Packs/Whois/Integrations/Whois/Whois.py (hunk 2 above)
Comment (Bargenish) [reply Packs/Whois/Integrations/Whois/Whois.py:8261]: To add the first dot before the TLD that was found. Should I add a comment and example inside the code?

Comment (ronykoz) [reply Packs/Whois/Integrations/Whois/Whois.py:8261]: yes please

File: Packs/Whois/Integrations/Whois/Whois.py (hunk 3 above)
Comment (ronykoz) [reply Packs/Whois/Integrations/Whois/Whois.py:8259]: than i think it should be just 0, as you check the len, not list

Comment (Bargenish) [reply Packs/Whois/Integrations/Whois/Whois.py:8259]: I tried it also and it failed with `'int' object is not iterable`. if the list is empty it requires another list to compare and run max function on.

**Comment by ronykoz** [comment]
Just fix the build and merge

**Pull request merged**